.ci_support/migrations/spdlog114.yaml

```diff
@@ -5,4 +5,4 @@ __migrator:
   migration_number: 1
 migrator_ts: 1722204008.301365
 spdlog:
-- '1.14'
+- '1.15'
```

Comment:
I find it confusing that we've edited the 1.14 migration file to use 1.15. This migration file is safely superseded by `spdlog115.yaml`. 

Though I agree it is a bit confusing having multiple migration files in the same feedstock, so I'm fine with either 1) deleting `spdlog114.yaml` entirely, or 2) restoring it to its original contents

Reply:
Oops, my mistake -- thank you @jdblischak for the code review

Reply:
@jdblischak https://github.com/TileDB-Inc/tiledbsoma-feedstock/pull/250/commits/b8ca57986776d74ad2ff33becbf1efde94ab8991